src/reanimated2/Jest.js

```diff
@@ -0,0 +1,163 @@
+export let withReanimatedTimer;
+export let moveAnimationByTime;
+export let moveAnimationByFrame;
+export let getAnimatedStyle;
+
+export const SetUpTests = (userConfig = {}) => {
+  const expect = require('expect');
+  const config = {
+    fps: 60,
+    ...userConfig,
+  };
+
+  global.performance = {
+    now: () => Date.now(),
+  };
+
+  const checkValidation = (received) => {
+    let message = [];
+    let isValid = true;
+    if (!received.props.style) {
+      isValid = false;
+      message.push("Component not contains 'style' property");
+    }
+    if (!received.props.animatedStyle?.value) {
+      isValid = false;
+      message.push("Component not contains 'animatedStyle' property");
+    }
+    message = message.join('\n');
+
+    return { isValid, message };
+  };
+
+  const getCurrentStyle = (received) => {
+    return {
+      ...received.props.style,
+      ...received.props.animatedStyle.value,
+    };
+  };
+
+  const findStyleDiff = (current, expect, requireAllMatch) => {
+    const diffs = [];
+    let isEqual = true;
+    for (const property in expect) {
```

Comment:
We're looping on the prototype chain here, is this intended?

Reply:
Yes, this is intended. Do you see any problem with this solution?

Reply:
No, just using `for..of` is more common, wanted to make sure.